Sources/StreamChat/Utils/TextLinkDetector.swift

```diff
@@ -0,0 +1,57 @@
+//
+// Copyright © 2024 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+/// The url and position of a link in a text.
+public struct TextLink: Equatable {
+    public let url: URL
+    public let range: NSRange
+}
+
+/// An object responsible to detect and parse links in text.
+public class TextLinkDetector {
+    private let detector: NSDataDetector?
+
+    public init() {
+        let types: NSTextCheckingResult.CheckingType = [.link]
+        do {
+            detector = try NSDataDetector(types: types.rawValue)
+        } catch {
+            detector = nil
+            assertionFailure("Data detector failed to be created in \(self)")
+        }
+    }
+
+    /// Checks if the provided text contains links or not.
+    public func hasLinks(in text: String) -> Bool {
```

Comment:
Since these are public methods, we should document them properly (add `Parameters` and `Returns`).

Reply:
In this case they are self explanatory, that is why I didn't add them 🤔

Reply:
I know, but It's good for our DocC docs, I think they are still autogenerated on the Swift Package Index pages.

Reply:
I'll update this then 👍